app/models/site_status.rb

```diff
@@ -35,6 +39,27 @@ class SiteStatus < ApplicationRecord
     suspended: "S",
   }, _prefix: :status
 
+  aasm column: :status, enum: true do
+    state :new_status, initial: true
+    state :running
+    state :suspended
+    state :discontinued
+
+    after_all_transitions :update_publish_flag
+
+    event :start do
+      transitions from: %i[new_status suspended discontinued], to: :running
+    end
+
+    event :suspend do
+      transitions from: :running, to: :suspended
+    end
+  end
```

Comment:
Seeing as we have an `enum` defined below here, maybe we should move this `aasm` definition and the `update_publish_flag` method definition down a little?

Reply:
fixed